mypy/checker.py

```diff
@@ -917,7 +917,7 @@ def visit_assignment_stmt(self, s: AssignmentStmt) -> Type:
     def check_assignment(self, lvalue: Node, rvalue: Node, infer_lvalue_type: bool = True) -> None:
         """Type check a single assignment: lvalue = rvalue."""
         if isinstance(lvalue, TupleExpr) or isinstance(lvalue, ListExpr):
-            ltuple = cast(Union[TupleExpr, ListExpr], lvalue)
+            ltuple = lvalue
```

Comment:
How would you feel about killing `ltuple` and replacing it's uses with `lvalue`?